back-end/src/sciler/config/workingConfigTypes.go

```diff
@@ -0,0 +1,320 @@
+}
+
+// CheckConstraints is a method that checks types and comparator operators
+func (and AndCondition) CheckConstraints(config WorkingConfig) error {
+	for _, logic := range and.logics {
+		err := logic.CheckConstraints(config)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Resolve is a method that checks if a condition is met
+func (and AndCondition) Resolve(config WorkingConfig) bool {
+	result := true
+	for _, logic := range and.logics {
+		result = result && logic.Resolve(config)
+	}
+	return result
+}
+
+// OrCondition is an operator which implements the LogicalCondition interface
+type OrCondition struct {
+	logics []LogicalCondition
+}
+
+// CheckConstraints is a method that checks types and comparator operators
+func (or OrCondition) CheckConstraints(config WorkingConfig) error {
+	for _, logic := range or.logics {
+		err := logic.CheckConstraints(config)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Resolve is a method that checks if a condition is met
+func (or OrCondition) Resolve(config WorkingConfig) bool {
+	result := false
+	for _, logic := range or.logics {
+		result = result || logic.Resolve(config)
+	}
+	return result
+}
+
+// LogicalConstraint is an interface for operators and constraints
+type LogicalConstraint interface {
+	Resolve(condition Condition, config WorkingConfig) bool
+	CheckConstraints(condition Condition, config WorkingConfig) error
+}
+
+// AndConstraint is an operator which implement the LogicalConstraint interface
+type AndConstraint struct {
+	logics []LogicalConstraint
+}
+
+// CheckConstraints is a method that checks types and comparator operators
+func (and AndConstraint) CheckConstraints(condition Condition, config WorkingConfig) error {
+	for _, logic := range and.logics {
+		err := logic.CheckConstraints(condition, config)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Resolve is a method that checks if a constraint is met
+func (and AndConstraint) Resolve(condition Condition, config WorkingConfig) bool { // todo: make lazy
+	result := true
+	for _, logic := range and.logics {
+		result = result && logic.Resolve(condition, config)
+	}
+	return result
+}
+
+// OrConstraint is an operator which implement the LogicalConstraint interface
+type OrConstraint struct {
+	logics []LogicalConstraint
+}
+
+// CheckConstraints is a method that checks types and comparator operators
+func (or OrConstraint) CheckConstraints(condition Condition, config WorkingConfig) error {
+	for _, logic := range or.logics {
+		err := logic.CheckConstraints(condition, config)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Resolve is a method that checks if a constraint is met
+func (or OrConstraint) Resolve(condition Condition, config WorkingConfig) bool { // todo: make lazy
+	result := false
+	for _, logic := range or.logics {
+		result = result || logic.Resolve(condition, config)
+	}
+	return result
+}
+
+//todo resolve on constraint
```

Comment:
is this to be done in a different issue?

Reply:
No, this is already done, I removed the todo